File: dev/test_data/csharp/test_main/aas_core_meta.v3/input/snippets/Verification/qualifier_types_are_unique.cs
/// <summary>
/// Check that there are no duplicate <see cref="Aas.IQualifier.Type" />'s
/// in the <paramref name="qualifiers" />.
/// </summary>
public static bool QualifierTypesAreUnique(
    IEnumerable<Aas.IQualifier> qualifiers
)
{
    var typeSet = new HashSet<string>();
    foreach (var qualifier in qualifiers)
    {
        if (typeSet.Contains(qualifier.Type))
        {
            return false;
        }
        typeSet.Add(qualifier.Type);
    }
    return true;
}

File: dev/test_data/java/test_main/aas_core_meta.v3/input/snippets/Verification/qualifier_types_are_unique.java
/**
* Check that there are no duplicate {@link IQualifier#getType() type}'s
* in the qualifiers.
 * @param qualifiers the qualifiers
*/
public static boolean qualifierTypesAreUnique(
  Iterable<IQualifier> qualifiers
) {
  Objects.requireNonNull(qualifiers);

  Set<String> typeSet = new HashSet<>();
  for (IQualifier qualifier : qualifiers) {
    if (typeSet.contains(qualifier.getType())) {
      return false;
    }
    typeSet.add(qualifier.getType());
  }
  return true;
}
